src/transaction-runner.coffee

```diff
@@ -66,8 +66,22 @@ class TransactionRunner
 
         catch error
           if error instanceof chai.AssertionError
-            data.message = "Failed assertion in hooks: " + error.message
-            data.test?.status = 'fail'
+            message = "Failed assertion in hooks: " + error.message
+
+            data['results'] ?= {}
+            data['results']['general'] ?= []
+            data['results']['general'].push { severity: 'error', message: message }
+
+            data.message = message
```

Comment:
please be consistent in style. Either use `[` `]` everywhere, or use `dot-notation` where applicable (above or below in a lot of places).